### ISSUE TITLE
Add Wayland watcher

### ISSUE THREAD

**Issue by SUPERCILEX** [SUPERCILEX/clipboard-history#10]

**Comment by SUPERCILEX** [comment]
Resources:
- https://emersion.fr/blog/2020/wayland-clipboard-drag-and-drop/
- https://wayland.app/protocols/wlr-data-control-unstable-v1

**Comment by Artem-B** [comment]
Can x11 watcher be used until proper wayland watcher exists? Wayland seems to emulate primary/clipboard reasonably well for the legacy X11 apps.

**Comment by SUPERCILEX** [comment]
Yup, x11 should work just fine. The main reason to add a native Wayland implementation is performance.

**Comment by SUPERCILEX** [comment]
Work remaining:
- Support claiming selection ownership
- Automatically trigger a paste

Spec: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/staging/ext-data-control/ext-data-control-v1.xml?ref_type=heads

Seems like I should be able to create a virtual keyboard from https://wayland.app/protocols/virtual-keyboard-unstable-v1 and get the keymap from https://smithay.github.io/wayland-rs/wayland_client/protocol/wl_keyboard/enum.Event.html.
Though actually I probably need to research https://wayland.app/protocols/text-input-unstable-v3 because that seems like the correct way to insert text. Hmmm, but I don't think I could insert stuff like images as that requires the application to run its code?

**Comment by SUPERCILEX** [comment]
Finally done in https://github.com/SUPERCILEX/clipboard-history/releases/tag/0.9.2.
